### PR TITLE
YARN-10152. Fix findbugs warnings in hadoop-yarn-applications-mawo-core module

### PR DESCRIPTION
**Pull request by aajisaka**
JIRA: https://issues.apache.org/jira/browse/YARN-10152

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ trunk Compile Tests _ |
| -1 :x: |  mvninstall  |  23m 32s |  root in trunk failed.  |
| -1 :x: |  compile  |   0m 27s |  hadoop-yarn-applications-mawo-core in trunk failed.  |
| +1 :green_heart: |  checkstyle  |   0m 19s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 22s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  17m 15s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  trunk passed  |
| +0 :ok: |  spotbugs  |   0m 36s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| -1 :x: |  findbugs  |   0m 31s |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core in trunk has 3 extant findbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 14s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 14s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 22s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 34s |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core generated 0 new + 0 unchanged - 3 fixed = 0 total (was 3)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 18s |  hadoop-yarn-applications-mawo-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   |  63m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1852 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 6ecc99d886b4 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / cb3f3cc |
| Default Java | 1.8.0_242 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/artifact/out/branch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/artifact/out/branch-compile-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-applications_hadoop-yarn-applications-mawo_hadoop-yarn-applications-mawo-core.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/artifact/out/branch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-applications_hadoop-yarn-applications-mawo_hadoop-yarn-applications-mawo-core-warnings.html |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/testReport/ |
| Max. process+thread count | 307 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core/src/main/java/org/apache/hadoop/applications/mawo/server/worker/WorkerId.java [hunk 1]
```diff
@@ -111,6 +111,9 @@ public final void setWorkerId(final String localworkerId) {
    * Implememt equals method for WorkerId.
    */
   public final boolean equals(final Object o) {
```
Comment (adamantal) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core/src/main/java/org/apache/hadoop/applications/mawo/server/worker/WorkerId.java]: Can we also add `@Override` to the `toString()` method as well?

Comment (aajisaka) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core/src/main/java/org/apache/hadoop/applications/mawo/server/worker/WorkerId.java]: Thank you for your comment. Added `@Override` to the `toString()`, `write(DataOutput)`, and `readFields(DataInput)`.

**Comment by aajisaka** [comment]
> Also, some findbugs warnings are still here - can you check?

The following findbugs report is for the trunk before applying the patch.
> https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/artifact/out/branch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-applications_hadoop-yarn-applications-mawo_hadoop-yarn-applications-mawo-core-warnings.html

The following is the findbugs report after applying the patch:
https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/1/artifact/out/patch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-applications_hadoop-yarn-applications-mawo_hadoop-yarn-applications-mawo-core-warnings.html

(Note: The report URLs before applying the patch start with "branch", and the reports after applying the patch start with "patch".)

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 10s |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 20s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   0m 17s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  16m  2s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  trunk passed  |
| +0 :ok: |  spotbugs  |   0m 31s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| -1 :x: |  findbugs  |   0m 30s |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core in trunk has 3 extant findbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 14s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 14s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 27s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 16s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 33s |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core generated 0 new + 0 unchanged - 3 fixed = 0 total (was 3)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 19s |  hadoop-yarn-applications-mawo-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   |  61m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1852 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 504dfe5c01e6 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1c5d2f1 |
| Default Java | 1.8.0_242 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/2/artifact/out/branch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-applications_hadoop-yarn-applications-mawo_hadoop-yarn-applications-mawo-core-warnings.html |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/2/testReport/ |
| Max. process+thread count | 308 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-mawo/hadoop-yarn-applications-mawo-core |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1852/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by aajisaka** [comment]
Hi @adamantal , would you review this? Thanks.

**Comment by aajisaka** [comment]
Thanks @adamantal for the review. Merging this based on Eric's +1 on the jira.

**Pull request merged**